native/cocos/bindings/manual/jsb_global.cpp

```diff
@@ -1392,6 +1394,110 @@ static bool jsb_register_TextDecoder(se::Object *globalObj) {
+            args.push_back(se::Value(ATHERMAL_STATUS_NONE));
+            args.push_back(se::Value(ATHERMAL_STATUS_SHUTDOWN));
+            CC_ASSERT_EQ(vmCallback.vmId, se::ScriptEngine::getInstance()->getVMId());
+            if (vmCallback.cbFn && vmCallback.cbFn->isObject() && vmCallback.cbFn->toObject()->isFunction()) {
+                vmCallback.cbFn->toObject()->call(args, nullptr);
+            }
+        });
+    });
+    return true;
+}
+SE_BIND_PROP_SET(jsb_adpf_onThermalStatusChanged_set)
+
+static bool jsb_adpf_onThermalStatusChanged_get(se::State &state) { // NOLINT
+    if (!vmCallback.cbFn) {
+        state.rval().setUndefined();
+    } else {
+        state.rval().setObject(vmCallback.cbFn->toObject());
+    }
+    return true;
+}
+SE_BIND_PROP_GET(jsb_adpf_onThermalStatusChanged_get)
+
+static bool jsb_adpf_getThermalStatus(se::State &state) { // NOLINT
+    int statusInt = ADPFManager::getInstance().GetThermalStatus();
+    state.rval().setUint32(statusInt);
+    return true;
+}
+SE_BIND_PROP_GET(jsb_adpf_getThermalStatus)
+
+static bool jsb_adpf_getThermalStatusMin(se::State &state) { // NOLINT
+    state.rval().setUint32(ATHERMAL_STATUS_NONE);
+    return true;
+}
+SE_BIND_PROP_GET(jsb_adpf_getThermalStatusMin)
+static bool jsb_adpf_getThermalStatusMax(se::State &state) { // NOLINT
+    state.rval().setUint32(ATHERMAL_STATUS_SHUTDOWN);
+    return true;
+}
+SE_BIND_PROP_GET(jsb_adpf_getThermalStatusMax)
+
+static bool jsb_adpf_getThermalStatusNormalized(se::State &state) { // NOLINT
+    float statusNormalized = ADPFManager::getInstance().GetThermalStatusNormalized();
+    state.rval().setFloat(statusNormalized);
+    return true;
+}
+SE_BIND_PROP_GET(jsb_adpf_getThermalStatusNormalized)
+
+static bool jsb_adpf_getThermalHeadroom(se::State &state) { // NOLINT
+    float headroom = ADPFManager::getInstance().GetThermalHeadroom();
+    state.rval().setFloat(headroom);
+    return true;
+}
+SE_BIND_PROP_GET(jsb_adpf_getThermalHeadroom)
+
+static void jsb_register_ADPF(se::Object *ns) { // NOLINT
+    se::Value adpfObj{se::Object::createPlainObject()};
+    adpfObj.toObject()->defineProperty("thermalHeadroom", _SE(jsb_adpf_getThermalHeadroom), nullptr);
+    adpfObj.toObject()->defineProperty("thermalStatus", _SE(jsb_adpf_getThermalStatus), nullptr);
+    adpfObj.toObject()->defineProperty("thermalStatusMin", _SE(jsb_adpf_getThermalStatusMin), nullptr);
+    adpfObj.toObject()->defineProperty("thermalStatusMax", _SE(jsb_adpf_getThermalStatusMax), nullptr);
+    adpfObj.toObject()->defineProperty("thermalStatusNormalized", _SE(jsb_adpf_getThermalStatusNormalized), nullptr);
+    adpfObj.toObject()->defineProperty("thermalHeadroom", _SE(jsb_adpf_getThermalHeadroom), nullptr);
+    adpfObj.toObject()->defineProperty("onThermalStatusChanged", _SE(jsb_adpf_onThermalStatusChanged_get), _SE(jsb_adpf_onThermalStatusChanged_set));
+    ns->setProperty("adpf", adpfObj);
+}
+
+#endif // CC_PLATFORM_ANDROID
```

Comment:
Would it better to put all ADPF related codes into another .cpp file?

Reply:
Currently, all custom JSB extension attributes are placed in this file. I don't think it's necessary to open another file.

Reply:
As this file becomes bigger and bigger, it is not suitable to do as previous.